### PR TITLE
xDS v3 config file option

### PR DESCRIPTION
**Pull request by stevesloka**
Adds a `server.xds-server-version` configuration item which allows users to define which version of xDS to use when serving xDS resources to Envoy. 

Updates #1898

### CODE REVIEW

File: cmd/contour/serve.go [hunk 1]
```diff
@@ -677,7 +695,7 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 			log = log.WithField("insecure", true)
 		}
 
-		log.Info("started xDS server")
+		log.Infof("started xDS server type: %q version: %q", ctx.XDSServerType, xDSServerVersion)
```
Comment (skriss) [inline cmd/contour/serve.go:698]: Maybe just add the server type and version as structured fields? e.g. 

```
log.WithField("xdsServerType", ctx.XDSServerType).WithField("xdsVersion", xDSServerVersion).Info("started xDS server")
```

**Review by skriss — CHANGES_REQUESTED** [review]
Need to add to `site/docs/main/configuration.md` as well.

File: cmd/contour/serve.go (hunk 1 above)
Comment (stevesloka) [reply cmd/contour/serve.go:698]: I modeled after the PR to refactor config out to the `pkg` package @skriss.

Comment (skriss) [reply cmd/contour/serve.go]: I still prefer structured log fields but 🤷‍♂️

**Review by skriss — APPROVED** [review]
LGTM

**Review by youngnick — COMMENTED** [review]
LGTM

**Pull request merged**